Library/DiscUtils.Vhd/DynamicStream.cs

```diff
@@ -316,16 +313,13 @@ public override int Read(byte[] buffer, int offset, int count)
                         if (readFromParent)
                         {
                             _parentStream.Position = _position;
-                            StreamUtilities.ReadFully(_parentStream, buffer, offset + numRead, toRead);
+                            StreamUtilities.ReadExact(_parentStream, buffer, offset + numRead, toRead);
```

Comment:
Did not throw before, does now. Bytes read was not tracked, so I believe correct behavior is to throw on a short read.